pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -25,7 +28,211 @@
 from .optimization_utils import find_next_gate, fuse_rot_angles
 
 
-@transform
+# pylint: disable = too-many-statements
+@lru_cache
+def _get_plxpr_merge_rotations():
+    try:
+        # pylint: disable=import-outside-toplevel
+        from jax import make_jaxpr
+        from jax.core import Jaxpr
+
+        from pennylane.capture import PlxprInterpreter
+        from pennylane.capture.primitives import measure_prim
+        from pennylane.operation import Operator
+    except ImportError:  # pragma: no cover
+        return None, None
+
+    # pylint: disable=redefined-outer-name, too-few-public-methods
+    class MergeRotationsInterpreter(PlxprInterpreter):
+        """Plxpr Interpreter for applying the ``merge_rotations``
+        transform when program capture is enabled."""
+
+        def __init__(self, atol: Optional[float] = 1e-8, include_gates: Optional[list[str]] = None):
+            self._env = {}
+            self.atol = atol
+            self.include_gates = include_gates
+
+            # dict[wire (int), op (Operator)]
+            self.previous_ops = {}
+
+        def interpret_and_refresh_previous_ops(self, op: Operator):
+            """Interpret the previous_ops dictionary and add the operator to the previous_ops dictionary."""
+
+            # Use list(dict.fromkeys(...)) as opposed to a set to maintain deterministic order
+            previous_ops_on_wires = list(dict.fromkeys(self.previous_ops.get(w) for w in op.wires))
+
+            # Refresh the previous_ops dictionary with the current operator
+            for o in previous_ops_on_wires:
+                if o is not None:
+                    for w in o.wires:
+                        del self.previous_ops[w]
```

Comment:
If my above suggestion is used:
```suggestion
                for w in o.wires:
                    del self.previous_ops[w]
```